### PR TITLE
feature(update):set time limit, break loop, and separate camera file

### PR DESCRIPTION
**Pull request by protruser**
set time limit
- if camera opens after 10 seconds, program will terminate the camera.

break loop
- len(good_matches) is very low criteria, so I set 20 -> 50

separate camera file
- In camera_utils.py file, the function is only open the camera.
- In updatedDetectionGreyScale.py, the algrorithm exists

### CODE REVIEW

**Comment by gogog01-29-2021** [comment]
Nice Quantitative Update!
I will test it!

**Comment by protruser** [comment]
If you find a bug, you can notice anytime!

**Comment by Minhyuk12** [comment]
I tried it with your code, I think it's better!

File: src/test_demo/updatedDetectionGreyScale.py [hunk 1]
```diff
@@ -59,10 +65,19 @@
...
+    else:
+        print("Not matching")
 
     # Display the video feed with detected template
     cv2.imshow('Detected Template', frame_matches)
 
+
+    # 카메라가 10초가 지나면 종료
+    elapsed_time = time.time() - start_time
+    if elapsed_time > 10:
```
Comment (Minhyuk12) [inline src/test_demo/updatedDetectionGreyScale.py:78]: How about changing elasped time to 20sec not 10?
I think program can't figure out fake one until 10secs.

File: src/test_demo/updatedDetectionGreyScale.py [hunk 2]
```diff
@@ -13,11 +17,13 @@
...
 
-# Access the camera
-cap = cv2.VideoCapture(0)
-if not cap.isOpened():
-    print("Error: Could not open camera.")
-    exit()
+# Use the common camera opening function
+cap = open_camera()
+
+# 카메라 시작 시간 기록
```
Comment (Minhyuk12) [inline src/test_demo/updatedDetectionGreyScale.py:23]: Please change comment to English

File: src/test_demo/updatedDetectionGreyScale.py [hunk 3]
```diff
@@ -59,10 +65,19 @@
...
         print("Match found!")
+        break
+    else:
+        print("Not matching")
 
     # Display the video feed with detected template
     cv2.imshow('Detected Template', frame_matches)
 
+
+    # 카메라가 10초가 지나면 종료
```
Comment (Minhyuk12) [inline src/test_demo/updatedDetectionGreyScale.py:76]: As I mentioned before, you should change comment to English.

**Review by Minhyuk12 — CHANGES_REQUESTED** [review]
I recommended some parts of codes

File: src/test_demo/updatedDetectionGreyScale.py (hunk 1 above)
Comment (protruser) [reply src/test_demo/updatedDetectionGreyScale.py:78]: Okay, I will change it right now

File: src/test_demo/updatedDetectionGreyScale.py (hunk 2 above)
Comment (protruser) [reply src/test_demo/updatedDetectionGreyScale.py]: I changed it! you can see it now

File: src/test_demo/updatedDetectionGreyScale.py (hunk 3 above)
Comment (protruser) [reply src/test_demo/updatedDetectionGreyScale.py]: I also changed it like before

File: src/test_demo/updatedDetectionGreyScale.py [hunk 4]
```diff
@@ -1,8 +1,12 @@
+
+# updatedDetectionGreyScale.py
 import cv2
 import numpy as np
+import time  # time 모듈 추가
+from camera_utils import open_camera
 
 # Load the template image
-template = cv2.imread('../../images/won_1000.jpg', 0)
+template = cv2.imread('../../images/euro_5.jpg', 0)
```
Comment (Minhyuk12) [inline src/test_demo/updatedDetectionGreyScale.py:9]: We use korean bill now on not euro

**Review by Minhyuk12 — CHANGES_REQUESTED** [review]
Request for changing image

File: src/test_demo/updatedDetectionGreyScale.py (hunk 4 above)
Comment (protruser) [reply src/test_demo/updatedDetectionGreyScale.py]: I fixed it, so could you check it now?

Comment (gogog01-29-2021) [reply src/test_demo/updatedDetectionGreyScale.py]: I will check it

**Comment by mannmi** [comment]
I've restructured the file organization to accommodate the new unit tests. Please note that most of the existing code is outdated, so I might need to overwrite many changes in the next commit.

Branch: feat/image_template_matching i will be merging my changes after i have aproved this git pull request.

**Pull request merged**